### DISCUSSION TITLE
Frontend API create account with custom content field

### DISCUSSION THREAD

**Discussion by boostvolt** [swellstores/community#220] [Help] [answered]
We have created a custom content field for the `accounts` model and would like to assign a value to it when creating an account via the frontend API, preferably via SwellJS. So far we have not found a way to directly address custom content fields and currently have to store data via the metadata as a workaround. Can this please be implemented accordingly for the frontend API?

**Comment by sags95** [comment]
@boostvolt content_fields are read-only by default, but this guide explains how to enable these for write_access https://developers.swell.is/guides/opening-content-permissions so that you can update them with swell-js. For example, if I have a content_field named `custom_field` within the `/accounts` model. I would use the `swell.account.update()` method like so after following the steps in the guide:

```
await swell.account.update({
      custom_field: 'testing'
    });
```

**Reply by boostvolt** [reply]
Thank you for the feedback. Can this be directly included in the `swell.account.create` endpoint? Because it doesn't make much sense to create an account without a `custom_field` which is marked as required.

**Reply by sags95** [reply]
Yes it should be possible to include these fields when creating an account

```
await swell.account.create({
      email: "${email}",
      password: '${password},
      custom_field: "testing"
    });
```

**Reply by boostvolt** [reply]
Unfortunately not via SwellJS because the `AccountSnake` interface is not generic enough.

**Reply by sags95** [reply]
Are you seeing a particular error?

**Reply by boostvolt** [reply]
So first we had to adapt the `AccountSnake` interface because custom fields are not accepted by the interface. However, if we then want to set a value for a custom field, according to your example, the user will be created correctly with all data except for the custom field.

**Reply by logeshswell** [reply, marked as answer]
@boostvolt Yes, you're right!
`AccountSnake` isn't accepting `content` field while [productSnake](https://github.com/swellstores/swell-js/blob/master/types/product/snake.d.ts#L139) accepts it. So, we need to move `content` field inside [BaseModel](https://github.com/swellstores/swell-js/blob/master/types/index.d.ts#L62), so that you could create/update content fields in any of the interfaces.

Would you be comfortable to create a PR for this change?

Secondly, you should be creating an account as follows since you're using `content` field.
```ts
await swell.account.create({
  email: "${email}",
  password: '${password},
  content: {
    custom_field: "testing"
  }
});
```
